assets/src/BlockEditorValidation.js

```diff
@@ -0,0 +1,96 @@
+const { registerPlugin } = wp.plugins;
+const { PluginPrePublishPanel } = wp.editPost;
+const { dispatch } = wp.data;
+const {__} = wp.i18n;
+
+export const blockEditorValidation = () => {
+  registerPlugin( 'pre-publish-checklist', { render: PrePublishCheckList } );
+};
+
+const PrePublishCheckList = () => {
+  let lockPost = false;
+
+  let checkListMsg = '';
+  let classname    = '';
+
+  // "Analytics & Tracking" fields validation.
+  let $htmlbody   = $( 'html, body' );
+  let $toValidate = $( '[data-validation]' );
+
+  let analyticsErrorFlag   = false;
+
+  if ( $toValidate.length ) {
+    let $row = null;
+
+    function add_required($row) {
+      $row.find('select').css({'border-color': 'red'});
+    }
+
+    function remove_required($row) {
+      $row.find('select').css({borderColor: ''});
+    }
+
+    $toValidate.each(function () {
+      const $this = $(this);
+      const val = $this.val();
+      $row = $this.parents('.cmb-row');
+
+      if ($this.is('[type="button"]') || $this.is('.cmb2-upload-file-id')) {
+        return true;
+      }
+
+      // Apply validation only for campaign post types.
+      if ('campaign' === $('#post_type').val() && 'required' === $this.data('validation')) {
+        if ($row.is('.cmb-type-file-list')) {
+
+          const has_LIs = $row.find('ul.cmb-attach-list li').length > 0;
+
+          if (!has_LIs) {
+            add_required($row);
+            analyticsErrorFlag = true;
+          } else {
+            remove_required($row);
+            analyticsErrorFlag = false;
+          }
+
+        } else {
+          if (!val || 'not set' === val) {
+            add_required($row);
+            analyticsErrorFlag = true;
+          } else {
+            remove_required($row);
+            analyticsErrorFlag = false;
+          }
+        }
+      }
+    });
+
+    if ( analyticsErrorFlag ) {
+      lockPost = analyticsErrorFlag;
+      // Open "Analytics & Tracking" fields postbox, if closed.
+      const p4_postbox = $htmlbody.find('#p4_campaign_fields');
+      if (p4_postbox.hasClass('closed')) {
+        p4_postbox.removeClass('closed');
+      }
+      checkListMsg += __( 'Please check "Analytics & Tracking" section for required fields.', 'planet4-master-theme-backend' );
+      classname = 'p4-plugin-pre-publish-panel-error';
+    }
+  }
+
+  if ( lockPost === true ) {
```

Comment:
If we use `analyticsErrorFlag`, then it's guaranteed to be a boolean, so it could just be `if ( analyticsErrorFlag ) {`